src/core/reducers/addons.js

```diff
@@ -432,6 +439,74 @@ export default function addonsReducer(
       return state;
     }
 
+    case UPDATE_RATING_COUNTS: {
+      const { addonId, oldReview, newReview } = action.payload;
+
+      const addon = state.byID[`${addonId}`];
+      if (!addon) {
+        return state;
+      }
+
+      const { ratings } = addon;
+      let average = ratings ? ratings.average : 0;
+      let ratingCount = ratings ? ratings.count : 0;
+      let reviewCount = ratings ? ratings.text_count : 0;
+
+      let countForAverage = ratingCount;
+      if (average && countForAverage && oldReview && oldReview.score) {
+        // Begin by subtracting the old rating to reset the baseline.
+        const countAfterRemoval = countForAverage - 1;
+
+        if (countAfterRemoval === 0) {
```

Comment:
It took me a while to figure out that we could not end up with `countAfterRemoval < 0` because of the `if` and the way JavaScript interprets `0`. That seems weird to me, because we should not rely on that (even though it is convenient here).

Maybe you could write the following?

```
if (average && average > 0 && countForAverage && countForAverage > 0 && ...) {
```

I don't know.

Reply:
You mean it is surprising that JavaScript interprets `0` as `false`? That is not surprising to me. I think adding more conditions to the if statement would make the code a bit harder to read because you'd have to take the time to comprehend each condition.

Reply:
In most languages `0 != false`, so maybe add a comment or something so that the next person reading the code won't spend minutes trying to understand why `countAfterRemoval` cannot be negative.